com.unity.render-pipelines.high-definition/Documentation~/Custom-Pass-Creating.md

```diff
@@ -161,6 +161,11 @@ When Unity renders a material that is not supported by the current injection poi
 
 <a name="Custom-Renderers-Pass-shader"></a>
 
+## Object ID Custom Pass
+HDRP includes a specialization of a Draw Renderers Custom Pass that draws the objects in the scene with a unique color (Object ID). This pass has the same parameters as the Draw Renderers Custom Pass except the material override option, which is hidden because this pass uses a built-in material that draws the Object ID colors. The Object IDs are generated before the first execution of the pass occurs. If a user script generates new objects procedurally, then the method AssignObjectIDs should be called manually, to generate new IDs.
+
+If some customization is required to the way Object IDs are generated, then we recommend writing a new Draw Renderers Custom Pass with a Custom Renderers Pass shader that provides the required functionality. You can find more details regarding Custom Renderers Pass shaders in the next section.
+
```

Comment:
- Could we have a short example of the `AssignObjectIDs` implementation to generate new Object ID's here? 
- An example of a Custom Renderers Pass that changes how HDRP generates Object IDs would be helpful too!

Reply:
Current re-write: 

The **Object ID Custom Pass** draws GameObjects in the scene with a unique color (`Object ID`). 

This pass has the same parameters as the [Draw Renderers Custom Pass](#Draw-Renderers-Custom-Pass) except the **Material** property is hidden because this pass uses a built-in Material to draw the Object ID colors. 

HDRP generates Object IDs before executing this custom pass. If you use a script to generate new GameObjects procedurally, call `AssignObjectIDs`  to generate new Object IDs.

To customize the way Unity generates Object IDs, write a new Draw Renderers Custom Pass using a [Custom Renderers Pass](#Custom-Renderers-Pass-shader) shader.

Reply:
Thanks for the feedback Vic, I will shortly update the PR based on the suggestions and I will also add the missing examples.